core/src/test/java/hudson/UtilTest.java

```diff
@@ -248,7 +258,22 @@ public void testIsSymlink() throws IOException, InterruptedException {
         }
     }
 
-    @Test public void deleteFile() throws Exception {
+    @Test
+    public void testDeleteFile() throws Exception {
+        File d = tmp.getRoot();
+        File f = new File(d, "f");
+        try {
+            // Test: File is deleted
+            mkfiles(f);
+            Util.deleteFile(f);
+            assertFalse("f exists after calling Util.deleteFile", f.exists());
+        } finally {
+            Util.deleteRecursive(d);
```

Comment:
I'm merely following the pattern that was already established in the original code.
The original code did:
  File d = tmp.getRoot(); 
  try {
    File f = new File(d, "f");
    ...
so I followed the existing convention that had been established.
If you would prefer, I could do:
  File f = tmp.newFile("f");
instead.

Re: teardown - Agreed.  I'll remove that "finally" clause.  I introduced it because my previous changes seemed to cause a unit-test failure in "jenkins.widgets.BuildListTableTest.linksFromFolders" and so I was being extra-careful, but I suspect that the true cause of that unit-test failure is nothing to do with my changes.